client/js/compute.js

```diff
@@ -0,0 +1,578 @@
+      call: function(v, x) { return v = x.pop() }
+    },
+    {
+      name: 'reverse',
+      desc: 'returns an array after reversing the order of its elements',
+      sample: 'var a = ${x} reverse',
+      call: function(v, x) { return v = x.reverse() }
+    },
+    {
+      name: 'shift',
+      desc: 'returns an array after removing its first element',
+      sample: 'var a = ${x} shift',
+      call: function(v, x) { return v = x.shift() }
+    },
+    {
+      name: 'sort',
+      desc: 'returns an array after sorting its elements alphabetically',
+      sample: 'var a = ${x} sort',
+      call: function(v, x) { return v = x.sort() }
+    },
+    {
+      name: 'numericSort',
+      desc: 'returns an array after sorting its elements numerically',
+      sample: 'var a = ${x} numericSort',
+      call: function(v, x) { return v = x.sort((a, b) => a.toString().localeCompare(b, undefined, {numeric: true, ignorePunctuation: true})) }
+    },
+    {
+      name: 'findIndex',
+      desc: 'returns the index of the first element in the array that matches the given value y',
+      sample: 'var a = ${x} findIndex ${y}',
+      call: function(v, x, y) { return v = x.findIndex(y) }
+    },
+    {
+      name: 'join',
+      desc: 'returns the array x as a combined string with values separated by an optional separator y, defaulting to comma',
+      sample: 'var a = ${x} join ${y}',
+      call: function(v, x, y) { return v = (y !== undefined) ? x.join(y) : x.join() }
+    },
+    {
+      name: 'slice',
+      desc: 'returns elements between start index y an end index z of an array (or characters of a string), excluding the end (unless z is omitted)',
+      sample: 'var a = ${x} slice ${y} ${z}',
+      call: function(v, x, y, z) { return v = (z !== undefined) ? x.slice(y, z) : x.slice(y) }
+    },
+    {
+      name: 'splice',
+      desc: 'removes elements starting from index x from the array in variable a (optionally limited to only y elements)',
+      sample: 'var a = splice ${x} ${y}',
+      call: function(v, x, y) { if(y !== undefined) {v.splice(x, y)} else {v.splice(x)}; return v }
+    },
+    {
+      name: 'push',
+      desc: 'adds the given value as the last array element of the variable',
+      sample: 'var a = push ${x}',
+      call: function(v, x) { v.push(x); return v }
+    },
+    {
+      name: 'unshift',
+      desc: 'adds the given value as the first array element of the variable',
+      sample: 'var a = unshift ${x}',
+      call: function(v, x) { v.unshift(x); return v}
+    },
+    {
+      name: 'randInt',
+      desc: 'returns random integer in range between (and including) two numbers',
+      sample: 'var a = randInt ${x} ${y}',
+      call: function(v, x, y) { return v = Math.floor((Math.random() * (y - x + 1)) + x) }
+    },
+    {
+      name: 'randRange',
+      desc: 'returns random integer in range between two numbers but excluding the endpoint, optionally in z increments (defaults to 1)',
```

Comment:
"returns **a** random..."? Same for `randInt`.

Reply:
It seems like technically randRange could do anything randInt can do, but it is a little more complicated.  To get a random integer from 1 to 10 you would need to put "var a = randRange 1 11".  Yet the benefit is that it could also limit the numbers to specific increments (e.g. even or odd numbers if z = 2).  But yeah, they both produce random integers.

randInt may be worth keeping because it is simple and will likely be used by novice users.  randRange may be worth keeping because it can do more, and matches up with certain functions such as slice which excludes the index of the last parameter like randRange excludes the number of its last parameter.